crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -603,38 +619,76 @@ impl<T: PoolTransaction> AllTransactions<T> {
                     }
                 };
             }
-            // If there's a nonce gap, we can shortcircuit, because there's nothing to update.
+
+            // tracks the balance if the sender was changed in the block
+            let mut changed_balance = None;
+
+            // check if this is a changed account
+            if let Some(info) = changed_accounts.get(&id.sender) {
+                let ancestor = TransactionId::ancestor(id.nonce, info.state_nonce, id.sender);
+                // If there's no ancestor then this is the next transaction.
+                if ancestor.is_none() {
+                    tx.state.insert(TxState::NO_NONCE_GAPS);
+                    tx.state.insert(TxState::NO_PARKED_ANCESTORS);
+                    tx.cumulative_cost = U256::ZERO;
+                    if tx.transaction.cost > info.balance {
+                        // sender lacks sufficient funds to pay for this transaction
+                        tx.state.remove(TxState::ENOUGH_BALANCE);
+                    } else {
+                        tx.state.insert(TxState::ENOUGH_BALANCE);
+                    }
+                }
+
```

Comment:
i forget how all this works, i think we'll prob want to extract all this state machine logic back into a diagram / spec